src/Tests/Microsoft.Diagnostics.Monitoring.TestCommon/TaskExtensions.cs

```diff
@@ -40,5 +41,28 @@ public static async Task<T> SafeAwait<T>(this Task<T> task, ITestOutputHelper ou
             }
             return fallbackValue;
         }
+
+        public static async Task WithCancellation(this Task task, CancellationToken token)
+        {
+            CancellationTokenSource localCancellationTokenSource = new();
```

Comment:
This new CTS is not needed. You can create a linked one from just the `token` parameter and then cancel it in the finally.